packages/babylon/src/parser/statement.js

```diff
@@ -539,7 +576,20 @@ export default class StatementParser extends ExpressionParser {
     }
     this.next();
     node.object = this.parseParenExpression();
+
+    // For the smartPipelines plugin:
+    // Disable topic references from outer contexts within the function body.
+    // They are permitted in function default-parameter expressions, which are
+    // part of the outer context, outside of the function body.
+    const outerContextTopicState = this.readTopicContextState();
+    this.enterTopicForbiddingContext();
```

Comment:
I can do this. I’d combine the three methods into one `withTopicContextState(callback)` function, putting the code that had been between the three old methods into the callback argument.

Reply:
My first thought was to just have it save & reset the previous state internally, but that's probably better.

Reply:
Yes, I’d prefer to avoid putting more state into the parser itself than necessary—we don’t need a stack, for instance. Let’s use lexical closure here.

Reply:
:+1: I dig it.